yt_dlp/extractor/vueschool.py

```diff
@@ -0,0 +1,151 @@
+from .common import InfoExtractor
+from .vimeo import VimeoIE
+from ..utils import (
+    int_or_none,
+    get_element_by_class,
+    get_element_text_and_html_by_tag,
+    extract_attributes,
+    compat_HTMLParseError
+)
+
+
+class VueSchoolBaseIE(InfoExtractor):
+    def _get_course_info_from_lesson_slug(self, lesson_slug, id):
+        return self._download_json(f"https://vueschool.io/api/lessons/{lesson_slug}/widgetInfo", video_id=id)
+
+    def __flatten_list_elements(self, lists):
+        output = []
+        for list in lists:
+            output += list
+        return output
+
+    def _flat_lessons_from_chapters(self, chapters):
+        return self.__flatten_list_elements([chapter.get("lessons") for chapter in chapters])
```

Comment:
```suggestion
        return self.__flatten_list_elements([x for chapter in chapters for x in chapter.get('lessons')])
```

and remove `__flatten_list_elements`